src/sid/simulate.py

```diff
@@ -229,6 +234,12 @@ def get_simulate_func(
     indexers = _prepare_assortative_matching_indexers(
         initial_states, contact_models, group_codes_info
     )
+    infection_probability_multiplier = _prepare_infection_probability_multiplier(
```

Comment:
Since the infection probability multiplier can depend on the values of the parameter we cannot call this just once in `get_simulate_func`. For the same reason we construct the assortative matching probabilities [here](https://github.com/covid-19-impact-lab/sid/blob/87ddc8289139f1b539354345fc2eedd6dbf598e0/src/sid/simulate.py#L331).

Reply:
Done.